File: leetcode/122_Best_Time_to_Buy_and_Sell_Stock_II.py
from typing import List
from unittest import TestCase

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        for i in range(1, len(prices)):
            if prices[i] > prices[i - 1]:
                profit += prices[i] - prices[i - 1]
        return profit

class MaxProfitTest(TestCase):
    def test_1(self) -> None:
        self.assertEqual(Solution().maxProfit([7,1,5,3,6,4]), 7)

    def test_2(self) -> None:
        self.assertEqual(Solution().maxProfit([1,2,3,4,5]), 4)

    def test_3(self) -> None:
        self.assertEqual(Solution().maxProfit([7,6,4,3,1]), 0)

File: leetcode/122_Best_Time_to_Buy_and_Sell_Stock_II.rs
struct Solution {}

impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut profit = 0;
        for i in 1..prices.len() {
            if prices[i] > prices[i - 1]{
                profit += prices[i] - prices[i - 1] ;
            }
        }
        return profit;
    }
}

mod max_profit {
    use super::*;

    #[test]
    fn test_1() {
        assert_eq!(Solution::max_profit(vec![7,1,5,3,6,4]), 7);
    }

    #[test]
    fn test_2() {
        assert_eq!(Solution::max_profit(vec![1,2,3,4,5]), 4);
    }

    #[test]
    fn test_3() {
        assert_eq!(Solution::max_profit(vec![7,6,4,3,1]), 0);
    }

}
